File: 1512. Number of Good Pairs.cs
//1

public class Solution {
    public int NumIdenticalPairs(int[] nums) {
        if(nums.Length == 1)
        {
            return 0;
        }
        if(nums.Length == 2)
        {
            if(nums[0] == nums[1])
            {
                return 1;
            }
            return 0;
        }
        var count = new List<int []>();
        for(int i = 0 ; i < nums.Length - 1 ; i++)
        {
            for(int j = i + 1 ; j < nums.Length ; j++)
            {
                count.Add([nums[i], nums[j]]);
            }
        }
        int c = 0;
        foreach(var i in count)
        {
            if(i[0] == i[1])
            {
                c++;
            }
        }
        return c;
    }
}

//2

public class Solution {
    public int NumIdenticalPairs(int[] nums) {
        var count = new Dictionary<int, int>();
        foreach(int num in nums)
        {
            if(count.ContainsKey(num))
            {
                count[num]++;
            }
            else
            {
                count[num] = 1;
            }
        }
        int c = 0;
        foreach(var i in count)
        {
            c = c + (i.Value * (i.Value - 1))/2;
        }
        return c;
    }
}

File: 1512. Number of Good Pairs.js
// 1
var numIdenticalPairs = function(nums) {
    if(nums.length === 1){
        return 0;
    }
    if(nums.length === 2){
        if(nums[0] === nums[1]){
            return 1;
        }
        return 0;
    }
    let hSet = new Set();
    for(let i = 0 ; i < nums.length - 1 ; i++){
        for(let j = i + 1 ; j < nums.length ; j++){
            hSet.add([nums[i],nums[j]]);
        }
    }
    let count = 0;
    for(let pair of hSet){
        console.log(pair);
        if(pair[0] === pair[1]){
            count++;
        }
    }
    return count
};


// 2

var numIdenticalPairs2 = function(nums) {
    let hMap = new Map();
    for(let num of nums){
        if(hMap.has(num)){
            hMap.set(num, hMap.get(num) + 1);
        } else {
            hMap.set(num, 1);
        }
    }
    let count = 0;
    for(let d of hMap){
        count = count + (d[1] * (d[1] - 1))/2;
        console.log(d[1]);
    }
    console.log(count);
    return count
};

console.log(numIdenticalPairs([1,2,3,1,1,3]));